src/config/loader.py

```diff
@@ -0,0 +1,122 @@
+			for try_num in range(max_num_tries):
+				try:
+					r = requests.get(self.config_source)
+					r.raise_for_status()
+					config_contents = r.text
+					break
+				except:
+					if try_num == max_num_tries - 1:
+						raise
+					self.logger.exception(f"Error while downloading config from '{self.config_source}'.")
+					time.sleep(1 + try_num * 2)
+		else:
+			with open(self.config_source, 'r', encoding='utf-8') as f:
+				config_contents = f.read()
+
+		assert config_contents is not None
+		config_hash = hashlib.sha256(config_contents.encode('utf-8')).hexdigest()
+		if config_hash != self.config_hash:
+			self.logger.info("Loading configuration from '%s'.", self.config_source)
+			config: Config = yaml.safe_load(config_contents)
+
+			log_level = logging.getLevelName(config.get('log_level', 'INFO') or 'INFO')
+			self.logger.setLevel(log_level)
+
+			limit = config.get('same_comment_per_PR_per_run_limit')
+			if limit is None:
+				# Match documented limit.
+				config['same_comment_per_PR_per_run_limit'] = 20
+
+			requeue_config = config.get('requeue_config')
+			if requeue_config is None:
+				requeue_config = RequeueConfig(max_per_run=DEFAULT_MAX_REQUEUES_PER_RUN)
+				config['requeue_config'] = requeue_config
+			else:
+				if requeue_config.get('max_per_run') is None:
+					requeue_config['max_per_run'] = DEFAULT_MAX_REQUEUES_PER_RUN
+
+			reset_votes_after_changes = config.get('reset_votes_after_changes')
+			if reset_votes_after_changes is not None:
+				assert isinstance(reset_votes_after_changes, list), f"reset_votes_after_changes must be a list. Got: {reset_votes_after_changes} with type: {type(reset_votes_after_changes)}"
+				reset_votes_after_changes = set(map_vote(vote) for vote in reset_votes_after_changes)
+				assert all(vote is not None for vote in reset_votes_after_changes), f"reset_votes_after_changes must be a list of integers. Got: {reset_votes_after_changes}"
+				config['reset_votes_after_changes'] = reset_votes_after_changes # type: ignore
+
+			rules = config['rules']
+			for rule in rules:
+				for name in ('author',) + ATTRIBUTES_WITH_PATTERNS:
+					if pat := rule.get(f'{name}_pattern'):
+						rule[f'{name}_regex'] = re.compile(pat, re.DOTALL) # type: ignore
+				if (pat := rule.get('diff_pattern')) is not None:
+					rule['diff_regex'] = re.compile(pat, re.DOTALL)
+				if (pat := rule.get('path_pattern')) is not None:
+					rule['path_regex'] = re.compile(pat)
+
+				vote = rule.get('vote')
+				if isinstance(vote, str):
+					rule['vote'] = map_vote(vote)
+
+				if (rule_policy_checks := rule.get('policy_checks')) is not None:
+					for rule_policy_check in rule_policy_checks:
+						for evaluation_check in rule_policy_check['evaluation_checks']:
+							evaluation_check['json_path_'] = JSONPath(evaluation_check['json_path'])
+							if (pat := evaluation_check.get('pattern')) is not None:
+								evaluation_check['regex'] = re.compile(pat)
+						match_type = rule_policy_check.get('match_type')
```

Comment:
`
if (match_type := rule_policy_check.get('match_type')) is not None:
    rule_policy_check['match_type'] = MatchType(match_type)
else:
    rule_policy_check['match_type'] = MatchType.ANY
`

Reply:
I changed the is to use the walrus operator as suggested.
BTW I just moved this code from elsewhere.
I kept the consequent and the alternate statements the same because I want to have the more typical case first.